### ISSUE TITLE
Can I somehow reduce time dimension to make my life easier? 

### ISSUE THREAD

**Issue by KarolinaChalupova** [KarolinaChalupova/DiplomaThesis#4]
Right now, the same model is trained multiple times (call it T times) on expanding window, there are essentially T models and T test corresponding test sets (same as Gu et al., 2018, we have discussed this @martinhronec). This makes sense from the perspective of using the model in practice for trading.
However, it is burdensome for purposes of interpretability. Given that I would like to try more architectures (simpler to more complex, as in Gu et al., 2018) and different seeds, there are a lot of models. So number of models is num of architectures times num of seeds tried times T. It would make my life easier if I could only have T=1, in other words, a single train-validation-test split for the whole data.  It would be a lot easier to code it as well as to handle the results. 
The question is: can I do it? There are two ways of doing this: 
- ASSUMING IT: getting permission to do it from prior literature, eg. Gu et al, 2018 - is the model stable in time? I need to look. 
- SHOWING IT: showing that the model is stable in time myself. This means showing e.g. that feature importance is stable in time.

@martinhronec any thoughts?

**Comment by martinhronec** [comment]
You can start without splits, i.e. just with one dataset. It could simply be some random sample from the whole dataset, but don't do T=1 only, this would be misleading, because model is **not** stable in time, both in terms of the hyperparameters, feature importance as well as the amount of predictability present.

Just do it for some sample (large enough) and then we will just loop it over multiple splits, this can be done later.

**Comment by barunik** [comment]
agree, I would also start with some random split, and then later check with multiple splits and see how it behaves

**Comment by KarolinaChalupova** [comment]
@martinhronec : ok. By multiple splits (that we will be doing later), do you mean multiple **random** (cross-sectional) splits or rolling splits **preserving time ordering** of data like in Kelly? Just so that I know how to write the code right now.

**Comment by KarolinaChalupova** [comment]
Why reduce time dimension: 
- Gu et al.: "The rank of important characteristics (top-third of the covariates ) is remarkably stable over time. This is true for all models." Top third means about 30 characteristics.
- great reduction in training time
- I will be reducing itime dimension for the purpose of presenting most of the results anyway (Kelly uses average, OTMH use average or choose single model)

So based on our discussion I think we can agree on following:
1. I'll start with a single random split, train multiple architectures and seeds and obtain preliminary results
2. Then I will check for SOME achitecture and seed that there is stability in time
3. Depending on what I find in 2., I will run more results for more splits in 1. 

So for now I will just train number of achitectures times number of seeds models and check the stability of results in the time dimension for some of these models.
